packages/mui-system/src/styleFunctionSx/styleFunctionSx.d.ts

```diff
@@ -57,7 +57,7 @@ export type SystemStyleObject<Theme extends object = {}> =
 export type SxProps<Theme extends object = {}> =
   | SystemStyleObject<Theme>
   | ((theme: Theme) => SystemStyleObject<Theme>)
-  | Array<SystemStyleObject<Theme> | ((theme: Theme) => SystemStyleObject<Theme>)>;
+  | Array<SystemStyleObject<Theme> | ((theme: Theme) => SystemStyleObject<Theme>) | boolean>;
```

Comment:
```suggestion
  | Array<boolean | SystemStyleObject<Theme> | ((theme: Theme) => SystemStyleObject<Theme>)>;
```

is it easier to read if boolean comes first?